bindings/python/mainloop.pyx

```diff
@@ -0,0 +1,13 @@
+
+cdef extern from "sol-mainloop.h":
+	int sol_init()
+	int sol_run()
+
+class SolettaMainloop():
+	def __init__(self):
+		sol_init()
+
+	def run(self):
+		sol_run()
```

Comment:
You should release the GIL before calling `sol_run()`, see http://docs.cython.org/src/userguide/external_C_code.html#releasing-the-gil

Then from C callbacks that would go back to Python, you must declare `with gil`. IOW, if you wrap `sol_timeout_add()`, the callback you give to that function is `cdef ... with gil`, so cython will emit preamble to acquire the interpreter lock before proceeding. If you follow Node.js and want to wrap `sol_platform_add_state_monitor()`, it is the same... or GPIO's `sol_gpio_config::in::cb`.